apps/navstar_overlay/src/navstar_overlay_netlink.erl

```diff
@@ -0,0 +1,170 @@
+%%%-------------------------------------------------------------------
+%% @doc navstar public API
+%% @end
+%%%-------------------------------------------------------------------
+-module(navstar_overlay_netlink).
+
+%% Application callbacks
+-export([ipneigh_replace/4, iproute_replace/5, bridge_fdb_replace/4,
+        iplink_show/2, iplink_add/5, iplink_set/3, iprule_show/1,
+        iprule_add/4, make_iprule/3, match_iprules/2, is_iprule_present/2,
+        ipaddr_replace/4, if_nametoindex/1]).
+
+-include_lib("gen_netlink/include/netlink.hrl").
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-define(TEST_AND_DEV, true).
+-endif.
+
+-ifdef(DEV).
+-define(TEST_AND_DEV, true).
+-endif.
+
+%% eg. ipneigh_replace(Pid, {44,128,0,1}, {16#70,16#b3,16#d5,16#80,16#00,16#03}, "vtep1024").
+ipneigh_replace(Pid, Dst, Lladdr, Ifname) ->
+  Attr = [{dst, Dst}, {lladdr, Lladdr}],
+  Neigh = {
+    _Family = inet,
+    _Ifindex = if_nametoindex(Ifname),
```

Comment:
In general, if you're doing something that has side-effects while initializing a Tuple, it's a bit difficult to track down when it breaks.